design/FY2019/GPU-Shading.md

```diff
@@ -0,0 +1,54 @@
+GPU-based Shading Calculations
+==============================
+
+**Neal Kruis, Big Ladder Software, LLC**
+
+## Justification for New Feature ##
+
+GPU architecture is optimized for performing geometry operations. Leveraging methodologies used in the gaming industry to represent shadows, the GPU can (in theory) improve computational performance of shading calculations and provide greater flexibility (e.g., handle concave geometries and solar transmission through interior glazing).
+
+## Approach ##
+
+The general approach is to use the Pixel Counting methodology described by Jones et al. In this approach, the building is rendered using OpenGL. Each surface is then viewed from the (parallel projection) perspective of the sun. The number of visible pixels for that surface is a proxy for the projected sunlit surface area. Dividing by the cosign of incidence for that surface gives the total sunlit surface area.
+
+### Exterior Shading ###
+
+Pixel counting for exterior shading can be handled using Big Ladder's [Penumbra](https://github.com/bigladder/penumbra) library (a C++ implementation of Jones's pixel counting method).
+
+### Interior Solar Distribution ###
+
+We will count pixels when viewing the from the perspective of the sun through a window with each internal surface assigned a different color, then use the histogramming functions in OpenGL to report the number of pixels of each color that are visible. This functionality will be added to Penumbra.
```

Comment:
Does this place a limit on the maximum number of surfaces (e.g. maximum number of colors)?

Reply:
How will this interior calculation work with blinds and shades on windows?

Reply:
I'll have to check, but I think we aren't limited to the number of colors. Even still, 255x255x255 = 16,581,375. Hard to imagine there would be more surfaces in a model (let alone a single zone).

Blinds/shades will work the same as they currently do. They aren't part of the shading calculations they affect the transmittance (of the solar through the window).

Scheduled transmittance is a different challenge discussed below in the document.